source/SAMRAI/pdat/SparseData.C

```diff
@@ -1291,6 +1288,24 @@ SparseData<BOX_GEOMETRY>::_removeInsideBox(
       }
    }
 }
+
+template <typename BOX_GEOMETRY>
+template <class T>
+inline void 
+SparseData<BOX_GEOMETRY>::hash_combine(std::size_t& seed, const T& v)
+{
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
+}
+
+template <typename BOX_GEOMETRY>
+template <class T>
+inline void
+SparseData<BOX_GEOMETRY>::to_lower(T& input)
```

Comment:
This seems like an odd method for a SparseData class.  It wasn't obvious why this needs to be here.  Maybe a comment would help.

Reply:
The `to_lower` method is used to lowercase all the strings being used as map keys. Boost provides a `to_lower`, but since we are removing boost we had to provide our own.